compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4586,6 +4586,14 @@ object Types {
   object MatchType {
     def apply(bound: Type, scrutinee: Type, cases: List[Type])(using Context): MatchType =
       unique(new CachedMatchType(bound, scrutinee, cases))
+
+    /** Extractor for `case _ =>` match type patterns */
+    object WildcardPattern {
```

Comment:
why that particular encoding?

Reply:
Could we just use a fresh TypeVar as the pattern type? Or `Any`? Are higher-kinded matches the problem here?